src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginUpdateHeader.java

```diff
@@ -0,0 +1,108 @@
+package com.github.manolo8.darkbot.gui.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.extensions.plugins.PluginHandler;
+import com.github.manolo8.darkbot.gui.components.MainButton;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicInteger;
+
+//todoo i18n
+public class PluginUpdateHeader extends JPanel {
+
+    private static final MessageFormat titleFormatter = new MessageFormat("Plugin Updates: {0} Last Checked: {1}");
+    private static final SimpleDateFormat dateFormatter = new SimpleDateFormat("EEE, d MMM, hh:mm a", Locale.ROOT);
+    private Date lastChecked = new Date();
+    private String status;
+
+    private final PluginHandler pluginHandler;
+
+    private final Title title;
+    private final UpdateAllButton updateAllButton;
+    private volatile UpdateProgressBar progressBar;
+
+    private final List<PluginName> names;
+
+    PluginUpdateHeader(Main main, List<PluginName> namess) {
```

Comment:
That moment when a gui element has a List of other GUI elements, but it's not to display them but rather to do logic on them

That logic MUST go away from PluginName (GUI) into a logic-based object